superduperdb/ext/transformers/training.py

```diff
@@ -344,11 +341,11 @@ def train(
     :param ray_configs: ray configs, must provide if using ray
     """
 
-    on_ray = on_ray or bool(ray_address) or bool(ray_configs)
+    on_ray = bool(ray_configs)
 
     # Auto detect multi-GPUs and use ray to run data parallel training
     # If not todo this, will run on a bad parallel mode
-    if not on_ray and torch.cuda.device_count() > 1:
+    if not on_ray and torch.cuda.device_count() > 1 and training_args.num_gpus != 1:
```

Comment:
when num_gpus is 0?

Reply:
This is to automatically start Ray for multi-GPU training in a multi-GPU environment, because currently all our multi-GPU training is based on Ray, whether or not Ray is connected as the computing layer. Therefore, there is no need to forcibly start Ray for single GPU and 0 GPU setups.